Some times we get a TypeError: Cannot read property 'length' of undefined

We are using isomorphic-git in Antora 2.3.4 where its package.json tells us: `"isomorphic-git": "0.78.5"` the error message we are seeing is shown as the following:

```
#!/bin/sh -eo pipefail
npm i -g antora-site-generator-lunr
DOCSEARCH_ENABLED=true DOCSEARCH_ENGINE=lunr NODE_PATH="$(npm -g root)" antora --generator antora-site-generator-lunr --stacktrace generate antora-playbook.yml

+ antora-site-generator-lunr@0.6.1
added 254 packages from 210 contributors in 11.703s
Error: Unknown TypeError: See cause (url: https://github.com/OpenNMS/opennms-js.git)
    at transformGitCloneError (/usr/local/lib/node_modules/antora-site-generator-lunr/node_modules/@antora/content-aggregator/lib/aggregate-content.js:794:22)
    at git.clone.then.then.catch (/usr/local/lib/node_modules/antora-site-generator-lunr/node_modules/@antora/content-aggregator/lib/aggregate-content.js:184:17)
Caused by: TypeError: Cannot read property 'length' of undefined
    at Inflate.push (/usr/local/lib/node_modules/antora-site-generator-lunr/node_modules/pako/lib/inflate.js:218:30)
    at listpack (/usr/local/lib/node_modules/antora-site-generator-lunr/node_modules/isomorphic-git/dist/for-node/isomorphic-git/index.js:3230:16)

Exited with code exit status 1

CircleCI received exit code 1
```
We can't really easily reproduce the problem, we just see it as a flapper in our CI/CD pipeline. The feedback we got from the Antora project was hinting to the isomorphic-git client: https://github.com/Mogztter/antora-site-generator-lunr/issues/97#issuecomment-874332413 dealing with timeouts and retries while dealing with repositories hosted on GitHub. If you have any hints for us on how we can mitigate this problem would nice.

Thank you in advance

--- a/src/utils/git-list-pack.js
+++ b/src/utils/git-list-pack.js
@@ -31,7 +31,7 @@ export async function listpack(stream, onData) {
     const inflator = new pako.Inflate()
     while (!inflator.result) {
       const chunk = await reader.chunk()
-      if (reader.ended) break
+      if (!chunk) break
       inflator.push(chunk, false)
       if (inflator.err) {
         throw new InternalError(`Pako error: ${inflator.msg}`)

